spec/streamflow/pm.md

```diff
@@ -0,0 +1,408 @@
+
+| Field                       | Type                         | Description                                                                               |
+| --------------------------- | ---------------------------- | ----------------------------------------------------------------------------------------- |
+| **fundsAtCreation**         | uint256                      | Funds that the reserve was created with                                                   |
+| **fundsRemaining**          | uint256                      | Funds remaining in the reserve taking into account funds claimed if the reserve is frozen |
+| **freezeRound**             | uint256                      | The round that the reserve is frozen because the sender overspent from its deposit        |
+| **recipientsInFreezeRound** | uint256                      | Number of recipients (i.e. registered orchestrators) during `freezeRound`                 |
+| **claimed**                 | mapping (address => uint256) | Amount claimed by an ETH address (only eligible recipients during `freezeRound`)          |
+
+A broadcaster only has a single reserve at any given point in time.
+
+A broadcaster's reserve can be thought of as a fixed amount of funds that is committed to the set of orchestrators which is updated each round. So, at the beginning of each round, the reserve is split into equal allocations based on the current orchestrator set. Thus, without adding additional funds to a reserve, it will be split into smaller allocations as the orchestrator set approaches its maximum size and it will be split into larger allocations as the orchestrator set shrinks in size. Since each allocation represents an orchestrator's max float for the broadcaster, as the allocation size increase, an orchestrator will be able to safely receive winning tickets with higher face values or more winning tickets with lower face values prior to having to redeem them.
+
+![reserve](images/reserve.png)
+
+## Global Parameters 
+
+| Field                      | Type    | Description                                                                                                |
+| -------------------------- | ------- | ---------------------------------------------------------------------------------------------------------- |
+| **UNLOCK_PERIOD**          | uint256 | The number of rounds that a sender must wait in order to unlock funds for withdrawal                       |
+| **FREEZE_PERIOD**          | uint256 | The number of rounds that a sender must wait after its reserve is frozen in order to add or withdraw funds |
+| **TICKET_VALIDITY_PERIOD** | uint256 | The number of rounds that a ticket is valid for starting from the ticket's `creationRound`                 |
+
```

Comment:
Would like to understand more about how the reserve is used by O to make decisions on whether to work with a specific B, and how we protect double-spend as the rounds progress (and # of O changes)

Reply:
> how the reserve is used by O to make decisions on whether to work with a specific B

I added a note about how orchestrators could accept or reject work from broadcasters based upon their reserves [here](https://github.com/livepeer/wiki/pull/36/files#diff-37a2c87fbfadd3ec58f7d2989b660216R76) along with an example. I think the concrete details for how work is accepted or rejected based on a broadcaster's reserve can be filled in when the client strategy is finalized.

> how we protect double-spend as the rounds progress (and # of O changes)

I tried to convey this in the [Reserve Claiming](https://github.com/livepeer/wiki/pull/36/files#diff-37a2c87fbfadd3ec58f7d2989b660216R354) section and in [this diagram](https://github.com/livepeer/wiki/pull/36/files#diff-37a2c87fbfadd3ec58f7d2989b660216R78). Do you think this needs to be clearer?

I added the following sentence "As rounds progress, a broadcaster's reserve is automatically committed to the registered orchestrators for the current round without any intervention by the broadcaster" at the [end of this paragraph](https://github.com/livepeer/wiki/pull/36/files#diff-37a2c87fbfadd3ec58f7d2989b660216R60).

I also added a note at the end of the Reserve Claiming section about [edge cases](https://github.com/livepeer/wiki/pull/36/files#diff-37a2c87fbfadd3ec58f7d2989b660216R385).

Reply:
Looks good to me.  I think the added notes addressed my questions here.